### ISSUE TITLE
Barons don't fight each other

### ISSUE THREAD

**Issue by srshelley** [Sesu8642/FeudalTactics#28]
I played on hard, loose, large map and the game ended in a 'stalemate' because the highest level units would not engage witg each other.  The other game allowed you to fight other 'barons' under certain conditions.

**Comment by Sesu8642** [comment]
Thanks for your input. Do you know how that works exactly?

**Comment by srshelley** [comment]
I think it is a function of distance, like Barons can only fight another
Baron if it is already their turn and they are only one space away.
I.e. they can't fight another Baron from anywhere on the map, only if the
Baron is in the hex space adjacent to them.
So to fight a Baron with a Baron your Baron will likely get killed in the
turn just prior, meaning you will need 2 Barons.  It makes the strategy
interesting.  But however you want to do it.

**Comment by Sesu8642** [comment]
Interesting. I try to replicate the original game mechanics pretty closely. But I don't actually own the original game :D
I guess the easiest way would be to make an attacking baron stronger than a defending one.

**Comment by d-albrecht** [comment]
I saw videos of the original game (haven't played it yet either) where the player commented that a stalemate could be possible in a specific situation, hence I'm not sure if you really could defeat units of the same strength. And if this was a thing, why would this only work with barons, shouldn't the same be true for all other units as well?  I feel like introducing exceptions to the rule just to prevent a stalemate is a poor design decision. If you happen to end up in a situation where no side can conquer the other, then this is one type of valid outcome to a match. Either this map seed just plainly isn't winnable, or you used the wrong strategy.

If levels of this game where pre-selected and vetted for being winnable, then you could ask for insights for how the game is winnable, but in a purely random generated seed it is just the reality that some matches will not be winnable and that's fine. And a few will end up in a stalemate.

And if this depends on the distance then no AI will give you a chance to come close enough to beat their baron, so you wouldn't prevent stalemates either.

I agree that it could be possible to define a general rule that would let you take tiles that are protected by some unit (meaning neighboring tiles, not the tile with the unit itself) with units of the same strength, but as I said, this should be a rule that applied to every level of strength and was well-documented in the tutorial.

**Comment by Sesu8642** [comment]
@d-albrecht interesting perspective on stalement/tie being a valid outcome. Maybe there should be a message box when it happens. But then I would need to detect it which is difficult because the situation can be resolved by one player moving their baron. It's not that important for singleplayer but could be something that needs to be handled when multiplayer happens.

**Comment by d-albrecht** [comment]
Well, in multi-player matches this should probably just come down to one player suggesting a draw (like in chess). And for single-player matches against AIs, there would be a lot to consider in order to detect a stalemate, so again, it's way easier if the human player just manually ends the match at some point. Or you could use similar rules like in chess, if both players (because in a stalemate you usually have only two players left; in general) don't move their units for a bunch of turns or only recreate the same set of positions, then the game could just detect that.

So, I would suggest: Don't bother with detecting actual stalemates, instead consider using general rules like in chess to end games with (a) the same position happening too often or (b) some amount of turns without changes in tile-ownership. This would work well in both single-player and multi-player mode and would force movement of any kind. Otherwise, I would rely on the human players to be "mature" enough to resign / offer a draw if an apparent stalemate happens.

**Comment by d-albrecht** [comment]
I can see the logic behind that suggestion but I'm not sure that this would actually improve the game in general. First of:
> it is possible for at least 2 players to have a baron in the field and still have a positive income

that hugely depends on the board size. So, all further argumentation would necessarily be tied to specific board sizes if that is the argument you want to make. Or in other words: differently sized games would offer a different number of units / unit upgrades. Given that each unit type needs a sprite, I'm not a fan of offering arbitrary number of units just to break a tie. But you are right that at least for extremely large boards, "only" having Barons can lead to more stalemates quickly. Even three-way stalemates.

Furthermore, it's a principle of this game that **creating** strong units is rather cheap (only 10 money per strength level), but **sustaining** them is getting more and more expensive (each level costs three times as much as the previous). Still, Barons are the first were the salary is higher than their cost. And if you aren't able to sustain the unit even one turn, then sure it will die, but you will never have to pay it's high salary. So, this game design - in my eyes - needs an upper bound / a cap of some sort. If upgrading would be the expensive part and sustaining was "relatively" cheap, I'd argue that then having more levels of strength could be an interesting idea. But promoting a play style where you upgrade the units just to have them die, feels odd to me. And if your opponent plays strategically, then you need two stronger units at least. Then it just becomes a game of who has the necessary money to overrun the opponent first. And if it more advisable to wait until you can create three or four stronger units to cut off another tile(s), or just two to move the border. And if both players have roughly the same income, the border will just be moved back and forth several times.

TL;DR: Given that extremely large map sizes were added, we probably could benefit from a stronger unit but I would be very careful with removing the strength cap entirely. And given that the AI loves to create unsustainable units all the time (in my experience at least), adding stronger units can quickly worsen the quality of the existing AIs.

**Comment by bb-froggy** [comment]
Detecting a stalemate could work like this: All players can attack only tiles protected by a baron and all players can afford their upkeep. AIs would offer to end the game as tie.

**Comment by visionok** [comment]
If there is no consensus on how to deal with a stalemate then I would settle for an indication whether the opposing side is losing income or not. If they are not losing income (ie they can maintain the stalemate indefinitely) then I can simply start a new game.

**Comment by Cwpute** [comment]
i'd go with the original way it was dealt with if i were to choose: https://github.com/Sesu8642/FeudalTactics/issues/28#issuecomment-1262685629

**Comment by Asmageddon** [comment]
This isn't quite a solution to stalemates, but what if attacking a baron with a baron locked them in combat for N turns, giving you the opportunity to fight over territory with it temporarily out of commission, hopefully able to cut it off from the kingdom?

**Comment by Cwpute** [comment]
Sounds interesting in theory, but unless the fighting lasts a single turn or until one of the barons dies (of starvation), i don't see this new rule being made easily understandable on a first encounter.
It's interesting but essentially boils down to rerdering both barons useless AND blocking two hex of space from any action in a single turn (one of them being enemy territory) while still having to pay for your baron's upkeep. What would prevent you from throwing the m both into another fight once the first one ends ? this doesn't make the game move forward enough imo.
